### PR TITLE
Fix WUT ThermoHygro with new hardware revisions

### PR DESCRIPTION
**Pull request by mpikzink**
ThermoHygro does not provide 'wtWebGraphThermoHygroPortName'

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hello @mpikzink 
It seems that your change is breaking existing (older) devices. 
Could you please add your test data to this PR, so we can check how to fix your situation without breaking current users ? 
Thanx

**Comment by mpikzink** [comment]
``./scripts/save-test-data.php -o wut -v 57720-os-143 --snmpsim`` does not create any file. Can someone help?

**Comment by PipoCanaja** [comment]
Hello
`./scripts/save-test-data.php -o wut -v 57720-os-143`
What error or output does it display ?

**Comment by mpikzink** [comment]
SNMP EngineID 0x80004fb8056d6f6e69746f722d646576656c6f70bda3d740, transportDomain (1, 3, 6, 1, 6, 1, 1, 0), transportAddress ('127.0.0.1', 43342), securityModel 2, securityName 359a67676f5c3e94c335cf25499e2704, securityLevel 1
Opening /opt/librenms/tests/snmpsim/wut_57720-os-143.snmprec controller
Request var-binds: 1.3.6.1.2.1.1.2.0=<>, flags: EXACT, GET
data error for /opt/librenms/tests/snmpsim/wut_57720-os-143.snmprec controller at 1.3.6.1.2.1.1.2.0, index broken?
Response var-binds: 1.3.6.1.2.1.1.2.0=<No Such Instance currently exists at this OID>
Using /opt/librenms/tests/snmpsim/wut_57720-os-143.snmprec controller selected by candidate b'359a67676f5c3e94c335cf25499e2704'; transport ID 1.3.6.1.6.1.1.0, source address 127.0.0.1, context name "b'359a67676f5c3e94c335cf25499e2704'"

thousand of times

**Comment by PipoCanaja** [comment]
No idea why it does not behave as expected. Here is the expected result of the command : 
```
librenms@librenms:~$ ./scripts/save-test-data.php -o wut -v 57720-os-143
Starting snmpsim listening on 127.1.6.1:1161... 
OS: wut
Module: all
Variant: 57720-os-143

Added device: 2
127.1.6.1 2 wut 
#### Load disco module core ####
OS: Web-Thermo-Hygrometer (wut)


>> Runtime for discovery module 'core': 0.0400 seconds with 459904 bytes
>> SNMP: [6/0.73s] MySQL: [5/0.01s]
#### Unload disco module core ####


#### Load disco module os ####
Location: <private>
Hardware: 
OS Version: 
OS Features: 
Serial: 

>> Runtime for discovery module 'os': 0.1570 seconds with 439504 bytes
>> SNMP: [4/0.02s] MySQL: [0/0.00s]
#### Unload disco module os ####


#### Load disco module ports ####


>> Runtime for discovery module 'ports': 0.1420 seconds with 5912 bytes
>> SNMP: [5/0.14s] MySQL: [1/0.00s]
#### Unload disco module ports ####


#### Load disco module ports-stack ####


>> Runtime for discovery module 'ports-stack': 0.0220 seconds with 1960 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload disco module ports-stack ####


#### Load disco module entity-physical ####

Caching OIDs: entPhysicalEntry

>> Runtime for discovery module 'entity-physical': 0.0240 seconds with 8112 bytes
>> SNMP: [1/0.02s] MySQL: [1/0.00s]
#### Unload disco module entity-physical ####

Module [ entity-state ] disabled globally.


#### Load disco module processors ####
Host Resources: UCD Resources: 

>> Runtime for discovery module 'processors': 0.0460 seconds with 143640 bytes
>> SNMP: [2/0.04s] MySQL: [2/0.00s]
#### Unload disco module processors ####


#### Load disco module mempools ####


>> Runtime for discovery module 'mempools': 0.0660 seconds with 95776 bytes
>> SNMP: [2/0.04s] MySQL: [0/0.00s]
#### Unload disco module mempools ####


#### Load disco module cisco-vrf-lite ####

>> Runtime for discovery module 'cisco-vrf-lite': 0.0010 seconds with 2648 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload disco module cisco-vrf-lite ####

Module [ cisco-mac-accounting ] disabled globally.

Module [ cisco-pw ] disabled globally.

Module [ vrf ] disabled globally.

Module [ cisco-cef ] disabled globally.

Module [ cisco-sla ] disabled globally.

Module [ cisco-cbqos ] disabled globally.

Module [ cisco-otv ] disabled globally.

Module [ cisco-qfp ] disabled globally.


#### Load disco module ipv4-addresses ####
!

>> Runtime for discovery module 'ipv4-addresses': 0.0460 seconds with 5408 bytes
>> SNMP: [2/0.04s] MySQL: [1/0.00s]
#### Unload disco module ipv4-addresses ####


#### Load disco module ipv6-addresses ####


>> Runtime for discovery module 'ipv6-addresses': 0.0510 seconds with 3520 bytes
>> SNMP: [2/0.05s] MySQL: [1/0.00s]
#### Unload disco module ipv6-addresses ####

Module [ route ] disabled globally.


#### Load disco module sensors ####
Caching data: sensors 
 ENTITY-SENSOR: Caching OIDs: entPhysicalDescr
Airflow: 
Current: 
Charge: 
Dbm: 
Fanspeed: 
Frequency: 
Humidity: Cur 40.8, Low: null, Low Warn: null, Warn: null, High: null
+
Load: 
Loss: 
Power: 
Power_consumed: 
Power_factor: 
Runtime: 
Signal: 
State: 
Count: 
Temperature: Cur 20.8, Low: null, Low Warn: null, Warn: null, High: null
+
Tv_signal: 
Voltage: 
Snr: 
Pressure: 
Cooling: 
Delay: 
Quality_factor: 
Chromatic_dispersion: 
Ber: 
Eer: 
Waterflow: 
Percent: 

>> Runtime for discovery module 'sensors': 0.2060 seconds with 88560 bytes
>> SNMP: [4/0.14s] MySQL: [35/0.05s]
#### Unload disco module sensors ####


#### Load disco module storage ####
aix_filesystem : hrStorage : 

>> Runtime for discovery module 'storage': 0.1170 seconds with 15696 bytes
>> SNMP: [5/0.10s] MySQL: [1/0.00s]
#### Unload disco module storage ####


#### Load disco module hr-device ####


>> Runtime for discovery module 'hr-device': 0.0480 seconds with 3624 bytes
>> SNMP: [2/0.05s] MySQL: [1/0.00s]
#### Unload disco module hr-device ####


#### Load disco module discovery-protocols ####
 CISCO-CDP-MIB: 
 LLDP-MIB: 
 OSPF Discovery: 

>> Runtime for discovery module 'discovery-protocols': 0.0550 seconds with 11960 bytes
>> SNMP: [0/0.00s] MySQL: [3/0.01s]
#### Unload disco module discovery-protocols ####


#### Load disco module arp-table ####


>> Runtime for discovery module 'arp-table': 0.0520 seconds with 2408 bytes
>> SNMP: [0/0.00s] MySQL: [2/0.00s]
#### Unload disco module arp-table ####

Module [ discovery-arp ] disabled globally.

Module [ junose-atm-vp ] disabled globally.


#### Load disco module bgp-peers ####
AS2 Updated AS 

>> Runtime for discovery module 'bgp-peers': 0.0410 seconds with 6080 bytes
>> SNMP: [2/0.04s] MySQL: [3/0.00s]
#### Unload disco module bgp-peers ####


#### Load disco module vlans ####
IEEE8021-Q-BRIDGE-MIB VLANs: 

>> Runtime for discovery module 'vlans': 0.0530 seconds with 9768 bytes
>> SNMP: [2/0.05s] MySQL: [1/0.00s]
#### Unload disco module vlans ####

Module [ vmware-vminfo ] disabled globally.

Module [ libvirt-vminfo ] disabled globally.

Module [ printer-supplies ] disabled globally.


#### Load disco module ucd-diskio ####


>> Runtime for discovery module 'ucd-diskio': 0.0210 seconds with 3224 bytes
>> SNMP: [1/0.02s] MySQL: [1/0.00s]
#### Unload disco module ucd-diskio ####

Module [ applications ] disabled globally.


#### Load disco module services ####

>> Runtime for discovery module 'services': 0.0000 seconds with 768 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module services ####


#### Load disco module stp ####


>> Runtime for discovery module 'stp': 0.0170 seconds with 5888 bytes
>> SNMP: [1/0.02s] MySQL: [1/0.00s]
#### Unload disco module stp ####


#### Load disco module ntp ####

>> Runtime for discovery module 'ntp': 0.0010 seconds with 280 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module ntp ####

Module [ loadbalancers ] disabled globally.

Module [ mef ] disabled globally.


#### Load disco module wireless ####

>> Runtime for discovery module 'wireless': 0.0140 seconds with 131480 bytes
>> SNMP: [0/0.00s] MySQL: [24/0.01s]
#### Unload disco module wireless ####


#### Load disco module fdb-table ####


>> Runtime for discovery module 'fdb-table': 0.0530 seconds with 3208 bytes
>> SNMP: [0/0.00s] MySQL: [2/0.00s]
#### Unload disco module fdb-table ####


#### Load disco module mpls ####

>> Runtime for discovery module 'mpls': 0.0020 seconds with 23952 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module mpls ####

Discovered in 2.327 seconds


Hostname:    127.1.6.1
Device ID:   2
OS:          wut


Created directory : /opt/librenms/rrd/127.1.6.1/

#### Load poller module core ####
Uptime:  27 days 10 hours 57 minutes 7 seconds

>> Runtime for poller module 'core': 0.0308 seconds with 23120 bytes
>> SNMP: [2/0.03s] MySQL: [0/0.00s]
#### Unload poller module core ####


#### Load poller module availability ####
Availability: 
1 day : 100%
7 days : 100%
30 days : 100%
365 days : 100%

>> Runtime for poller module 'availability': 0.0203 seconds with 38136 bytes
>> SNMP: [0/0.00s] MySQL: [9/0.01s]
#### Unload poller module availability ####

Module [ unix-agent ] disabled globally.


#### Load poller module os ####
Generic :(
Location: <private>
Hardware: 
OS Version: 
OS Features: 
Serial: 

>> Runtime for poller module 'os': 0.0021 seconds with 552 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module os ####


#### Load poller module ipmi ####

>> Runtime for poller module 'ipmi': 0.0016 seconds with 2352 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module ipmi ####


#### Load poller module sensors ####
Checking (snmp) humidity rel. Feuchte... 
40.8 %
Checking (snmp) temperature Temperatur... 
20.8 C

>> Runtime for poller module 'sensors': 0.0326 seconds with 2648 bytes
>> SNMP: [2/0.03s] MySQL: [3/0.00s]
#### Unload poller module sensors ####


#### Load poller module processors ####

>> Runtime for poller module 'processors': 0.0016 seconds with 1288 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module processors ####


#### Load poller module mempools ####

>> Runtime for poller module 'mempools': 0.0013 seconds with 304 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module mempools ####


#### Load poller module storage ####

>> Runtime for poller module 'storage': 0.0018 seconds with 1496 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module storage ####


#### Load poller module netstats ####
 ICMP IP IP-FORWARD SNMP TCP TCPHC UDP

>> Runtime for poller module 'netstats': 0.1384 seconds with 3128 bytes
>> SNMP: [7/0.13s] MySQL: [0/0.00s]
#### Unload poller module netstats ####


#### Load poller module hr-mib ####


>> Runtime for poller module 'hr-mib': 0.0155 seconds with 1608 bytes
>> SNMP: [1/0.01s] MySQL: [0/0.00s]
#### Unload poller module hr-mib ####


#### Load poller module ucd-mib ####

>> Runtime for poller module 'ucd-mib': 0.0428 seconds with 1864 bytes
>> SNMP: [2/0.04s] MySQL: [0/0.00s]
#### Unload poller module ucd-mib ####


#### Load poller module ipSystemStats ####


>> Runtime for poller module 'ipSystemStats': 0.0252 seconds with 1904 bytes
>> SNMP: [1/0.02s] MySQL: [0/0.00s]
#### Unload poller module ipSystemStats ####


#### Load poller module ports ####
Caching Oids: Full ports polling dot3StatsDuplexStatus

>> Runtime for poller module 'ports': 0.1136 seconds with 23240 bytes
>> SNMP: [5/0.11s] MySQL: [1/0.00s]
#### Unload poller module ports ####


#### Load poller module customoid ####

>> Runtime for poller module 'customoid': 0.0023 seconds with 2488 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module customoid ####


#### Load poller module bgp-peers ####

>> Runtime for poller module 'bgp-peers': 0.0040 seconds with 12128 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module bgp-peers ####

Module [ junose-atm-vp ] disabled globally.

Module [ printer-supplies ] disabled globally.


#### Load poller module ucd-diskio ####

>> Runtime for poller module 'ucd-diskio': 0.0013 seconds with 1672 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module ucd-diskio ####

Module [ wifi ] disabled globally.


#### Load poller module wireless ####

>> Runtime for poller module 'wireless': 0.0014 seconds with 1288 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module wireless ####


#### Load poller module ospf ####
 Processes: 0
>> Runtime for poller module 'ospf': 0.0258 seconds with 31352 bytes
>> SNMP: [1/0.02s] MySQL: [0/0.00s]
#### Unload poller module ospf ####

Module [ cisco-ipsec-flow-monitor ] disabled globally.

Module [ cisco-remote-access-monitor ] disabled globally.

Module [ cisco-cef ] disabled globally.

Module [ cisco-sla ] disabled globally.

Module [ cisco-mac-accounting ] disabled globally.

Module [ cipsec-tunnels ] disabled globally.

Module [ cisco-ace-loadbalancer ] disabled globally.

Module [ cisco-ace-serverfarms ] disabled globally.

Module [ cisco-asa-firewall ] disabled globally.

Module [ cisco-voice ] disabled globally.

Module [ cisco-cbqos ] disabled globally.

Module [ cisco-otv ] disabled globally.

Module [ cisco-qfp ] disabled globally.

Module [ cisco-vpdn ] disabled globally.

Module [ nac ] disabled globally.

Module [ netscaler-vsvr ] disabled globally.

Module [ aruba-controller ] disabled globally.


#### Load poller module entity-physical ####

>> Runtime for poller module 'entity-physical': 0.0023 seconds with 2064 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module entity-physical ####

Module [ entity-state ] disabled globally.


#### Load poller module applications ####

>> Runtime for poller module 'applications': 0.0016 seconds with 1592 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module applications ####

Module [ mib ] disabled globally.


#### Load poller module stp ####


>> Runtime for poller module 'stp': 0.0154 seconds with 2608 bytes
>> SNMP: [1/0.01s] MySQL: [1/0.00s]
#### Unload poller module stp ####


#### Load poller module ntp ####

>> Runtime for poller module 'ntp': 0.0011 seconds with 368 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module ntp ####

Module [ loadbalancers ] disabled globally.

Module [ mef ] disabled globally.


#### Load poller module mpls ####

>> Runtime for poller module 'mpls': 0.0004 seconds with 160 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module mpls ####

Enabling graphs: +++++++++

Polled in 1.528 seconds
Saved to /opt/librenms/tests/data/wut_57720-os-143.json
Ready for testing!
```

**Comment by PipoCanaja** [comment]
Now we'll check if your change does impact the regression tests. If yes, we'll have to check if the impact is negative or not, and regenerate the regression test data accordingly (or change the PR to ensure we don't break anything).

**Comment by PipoCanaja** [comment]
And it does indeed break. 
You cannot remove the OIDs starting with `1.3.6.1.4.1.5040.1.2.9` as they are the only OIDs supported by this device : 
` tests/snmpsim/wut_57613.snmprec `.
So you'll have to update your YAML file and add your OIDs in `1.3.6.1.4.1.5040.1.2.42` but leave the existing `1.3.6.1.4.1.5040.1.2.9` as well.

File: includes/definitions/discovery/wut.yaml [hunk 1]
```diff
@@ -45,14 +45,14 @@ modules:
                 -
                     oid: wtWebGraphThermoHygro
                     value: wtWebGraphThermoHygroBinaryTempValue
-                    num_oid: '.1.3.6.1.4.1.5040.1.2.9.1.4.1.1.{{ $index }}'
```
Comment (mpikzink) [inline includes/definitions/discovery/wut.yaml:48]: mistake in my previous commit. The old revision answers on ...9... & ...42...

**Comment by PipoCanaja** [comment]
Your code still breaks current non-regression tests from other devices. Meaning your change will impact existing users of these devices. Could you check the test results and fix it (so current users are not impacted) ?

File: includes/definitions/discovery/wut.yaml [hunk 2]
```diff
@@ -20,13 +20,13 @@ modules:
                     oid: wtWebGraphThermoHygro
                     value: wtWebGraphThermoHygroBinaryTempValue
                     num_oid: '.1.3.6.1.4.1.5040.1.2.42.1.4.1.1.{{ $index }}'
-                    descr: wtWebGraphThermoHygroPortName
+                    descr: 'Temperatur'
```
Comment (Jellyfrog) [inline includes/definitions/discovery/wut.yaml:23]: This doesn't seem correct. Isnt the old Oid available? In other case it's misspelled

Comment (mpikzink) [reply includes/definitions/discovery/wut.yaml:23]: No, this was the reason why I started the PR :-). It is German spelled as on the WebUI (German vendor) and before. But in the meantime I think environment is more clear.

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**